client/app/reducers/selection.coffee

```diff
@@ -0,0 +1,38 @@
+Immutable = require 'immutable'
+
+{ActionTypes} = require '../constants/app_constants'
+
+DEFAULTSTATE =
+    selected: Immutable.Set()
+    allSelected: false
+
+module.exports = (state = DEFAULTSTATE, action) ->
+    switch action.type
+        when ActionTypes.MAILBOX_SELECT_ALL
+            if state.allSelected
+                nextState = DEFAULTSTATE
+            else
+                nextState =
+                    allSelected: true,
+                    selected: Immutable.Set()
+
+        when ActionTypes.MAILBOX_SELECT
+            id = action.value.id
+            if(action.value.value and not state.selected.has(id))
+                nextState =
+                    allSelected: state.allSelected,
```

Comment:
Don't `allSelected` should be false except when `id` was the only message not selected ?